sdk/nodejs/automation/stack.ts

```diff
@@ -535,13 +536,50 @@ Event: ${line}\n${e.toString()}`);
             });
         }
 
-        const kind = this.workspace.program ? execKind.inline : execKind.local;
+        let onExit = (hasError: boolean) => {
+            return;
+        };
+        let didError = false;
+
+        let kind = execKind.local;
+        if (this.workspace.program !== undefined) {
+            const ver = semver.parse(this.workspace.pulumiVersion) ?? semver.parse("3.0.0")!;
+            // 3.181 added support for --client (https://github.com/pulumi/pulumi/releases/tag/v3.181.0)
+            if (ver.compare("3.181.0") < 0) {
```

Comment:
I might be reading the docs wrong but I think there may be `ver.lt("3.181.0")`, which'd be easier to read/harder to get wrong? Also as above (and in the Python also) I think we should consider factoring the check (or at least the version number) out?